doc/sphinx/explore-mir-using-demos.md

```diff
@@ -0,0 +1,133 @@
+# Explore What Mir Is Capable of Using Mir Demos
```

Comment:
I would rename this to "How to Run Mir Demos" and move it to the "How-Tos", but I know Alan has thoughts about this XD

Reply:
Think from the reader's perspective: The point is learning about what Mir does, it is not about running the Mir demos

Reply:
Having thought some more, this is also teaching about the ways to deploy Mir based compositors